starknet_py/tests/e2e/proxy_check/proxy_test.py

```diff
@@ -1,140 +1,171 @@
+from typing import Optional
+
 import pytest
+from starkware.starknet.public.abi import (
+    get_storage_var_address,
+)
 
 from starknet_py.contract import Contract
+from starknet_py.net.client_errors import ContractNotFoundError
+from starknet_py.proxy_check import ProxyResolutionError, ProxyCheck
+
 
-MAX_FEE = int(1e20)
+async def put_and_get_map_contract(map_contract: Contract, key: int, val: int) -> int:
+    """Put (key, val) into map_contract's storage and retrieve value under the key"""
+    await (
+        await map_contract.functions["put"].invoke(key, val, max_fee=int(1e16))
+    ).wait_for_acceptance()
+    (result,) = await map_contract.functions["get"].call(key=key)
+    return result
 
 
-@pytest.mark.skip
 @pytest.mark.asyncio
-async def test_argent_contract_from_address_throws_on_too_many_steps(
-    gateway_client, map_contract, compiled_proxy
+async def test_contract_from_address_no_proxy(
+    gateway_account_client, map_compiled_contract
 ):
-    proxy1_deployment = await Contract.deploy(
-        compiled_contract=compiled_proxy,
-        constructor_args=[map_contract.address],
-        client=gateway_client,
-    )
-    proxy2_deployment = await Contract.deploy(
-        compiled_contract=compiled_proxy,
-        constructor_args=[proxy1_deployment.deployed_contract.address],
-        client=gateway_client,
+    deployment_result = await Contract.deploy(
+        compiled_contract=map_compiled_contract,
+        client=gateway_account_client,
     )
+    await deployment_result.wait_for_acceptance()
 
-    await proxy1_deployment.wait_for_acceptance()
-    await proxy2_deployment.wait_for_acceptance()
-
-    with pytest.raises(RecursionError) as exinfo:
-        await Contract.from_address(
-            proxy2_deployment.deployed_contract.address,
-            client=gateway_client,
-            proxy_config={"max_steps": 2},
-        )
-
-    assert "Max number of steps exceeded" in str(exinfo.value)
-
-
-async def set_implementation(proxy1: Contract, proxy2: Contract):
-    argent_proxy = "_set_implementation" in proxy1.functions
-    set_implementation_name = (
-        "_set_implementation" if argent_proxy else "set_implementation_hash"
+    contract = await Contract.from_address(
+        address=deployment_result.deployed_contract.address,
+        client=gateway_account_client,
     )
-    implementation = "implementation" if argent_proxy else "new_implementation"
 
-    params = {
-        "proxy1": {implementation: proxy2.address},
-        "proxy2": {implementation: proxy1.address},
-    }
+    assert all(f in contract.functions for f in ("put", "get"))
```

Comment:
nitpick: assert knows how to compare different structures when using != and ==. You can use that to compare sets. This way when there is something missing it will let you know what. With current code you'll only know that returned value was false.

```
assert set(contract.functions.keys()) == {"put", "get"}
```